crypto_commons/netcat/netcat_commons.py

```diff
@@ -10,12 +10,12 @@ def nc(host, port):
 
 
 def receive_until(s, delimiters, break_on_empty=False):
-    all_data = ""
+    all_data = b""
     data = s.recv(1)
     while data not in delimiters:
         all_data += data
         data = s.recv(1)
-        if data == '' and break_on_empty:
+        if len(data) == 0 and break_on_empty:
```

Comment:
```suggestion
        if not data and break_on_empty:
```
Seems to be a bit more pythonic